services/blockchain-indexer/tests/unit/shared/dataservice/business/mainchain.test.js

```diff
@@ -0,0 +1,172 @@
+		const result = await isMainchain();
+		expect(typeof result).toBe('boolean');
+		expect(result).toBe(false);
+	});
+
+	it('should return false -> null chainID', async () => {
+		jest.mock(mockedFilePath, () => {
+			const actual = jest.requireActual('../../../../../shared/dataService/business/network');
+			return {
+				...actual,
+				getNetworkStatus() {
+					return { data: { chainID: null } };
+				},
+			};
+		});
+
+		const { isMainchain } = require('../../../../../shared/dataService');
+		const result = await isMainchain();
+		expect(typeof result).toBe('boolean');
+		expect(result).toBe(false);
+	});
+
+	it('should return true -> valid mainchain chainID', async () => {
+		jest.mock(mockedFilePath, () => {
+			const actual = jest.requireActual('../../../../../shared/dataService/business/network');
+			return {
+				...actual,
+				getNetworkStatus() {
+					return { data: { chainID: '04000000' } };
+				},
+			};
+		});
+
+		const { isMainchain } = require('../../../../../shared/dataService');
+		const result = await isMainchain();
+		expect(typeof result).toBe('boolean');
+		expect(result).toBe(true);
+	});
+
+	it('should return false -> valid sidechain chainID', async () => {
+		jest.mock(mockedFilePath, () => {
+			const actual = jest.requireActual('../../../../../shared/dataService/business/network');
+			return {
+				...actual,
+				getNetworkStatus() {
+					return { data: { chainID: '04000001' } };
+				},
+			};
+		});
+
+		const { isMainchain } = require('../../../../../shared/dataService');
+		const result = await isMainchain();
+		expect(typeof result).toBe('boolean');
+		expect(result).toBe(false);
+	});
+
+	it('should return false -> invalid chainID', async () => {
+		jest.mock(mockedFilePath, () => {
+			const actual = jest.requireActual('../../../../../shared/dataService/business/network');
+			return {
+				...actual,
+				getNetworkStatus() {
+					return { data: { chainID: 'invalid' } };
+				},
+			};
+		});
+
+		const { isMainchain } = require('../../../../../shared/dataService');
+		const result = await isMainchain();
+		expect(typeof result).toBe('boolean');
+		expect(result).toBe(false);
+	});
+});
+
+describe('Test resolveMainchainServiceURL method', () => {
+	it('should return devnet mainchain URL -> valid devnet mainchain chainID', async () => {
+		const chainID = '04000000';
+		jest.mock(mockedFilePath, () => {
+			const actual = jest.requireActual('../../../../../shared/dataService/business/network');
+			return {
+				...actual,
+				getNetworkStatus() {
+					return { data: { chainID } };
+				},
+			};
+		});
+
+		const { resolveMainchainServiceURL } = require('../../../../../shared/dataService');
+		const result = await resolveMainchainServiceURL();
+
+		const { serviceURL } = config.networks.LISK.find(c => chainID === c.chainID);
+		expect(result).toBe(serviceURL);
+	});
+
+	it('should return betanet mainchain URL -> valid devnet mainchain chainID', async () => {
```

Comment:
Also, isn't either this or the above devnet check sufficient to cover the case?

```suggestion
	it('should return betanet mainchain URL -> valid betanet mainchain chainID', async () => {
```